src/core/ext/transport/chaotic_good/frame_header.h

```diff
@@ -45,11 +47,11 @@ struct FrameSizes {
 };
 
 struct FrameHeader {
-  FrameType type;
-  BitSet<3> flags;
+  uint32_t type_and_flags;  // lower 8 bits specify the frame type.
```

Comment:
This approach will cause all frame manipulation code to know about this serialization detail, rather than just the code that's serializing it - I think it would be better to keep the type, flags separation (even if we need to change the flags representation)